### PR TITLE
Version bump CI tools

### PR DESCRIPTION
**Pull request by felixdivo**
The automated static analysis tools are updated to their most recent version.

EDIT: I basically updated the tools in `requirements-lint.txt` and some images in Travis CI. Then I and re-ran the formatter, which  causes much of the diff. I also removed the formatting job from Travis as it was already done by Github actions. I removed the normal testing from Travis, as it is much easier to specify in Github actions (and we now also test more platforms, like properly test on macOS!). We now also test on newer versions of Python as well (currently up to `3.10.0-alpha.4`) and on Pypy on all platforms (also new).

### CODE REVIEW

**Review by hardbyte — APPROVED** [review]
:medal_military:

**Comment by felixdivo** [comment]
How useful is it to have both Github actions as well as AppVeyor an Travis CI? Wouldn't it be nice to at least remove the tests from Travis CI as they seem to be very slow and a duplicate. Also, do we need AppVeyor at all?

**Comment by hardbyte** [comment]
> How useful is it to have both Github actions as well as AppVeyor an Travis CI? Wouldn't it be nice to at least remove the tests from Travis CI as they seem to be very slow and a duplicate. Also, do we need AppVeyor at all?
> 
> EDIT: Proposal:
> 
>     * retain all Github workflow tests, which test with CPython on Linux, macOS and Windows
> 
>     * Move CPython 3.10-dev & nightly from Travis to Github workflow
> 
>     * remove Travis CI testing except: pypy3 and socketcan tests
> 
>     * retain the other Travis jobs like the linter and publishing (for now at least, we can consider moving them in another PR)
> 
>     * remove AppVeyor testing as it is entirely covered by Github workflows (right?)

Travis-CI _might_ be a bit faster now I just [migrated](https://docs.travis-ci.com/user/migrate/open-source-repository-migration#frequently-asked-questions) to the travis-ci.com version.
But yes I think just relying on github workflow except where we need additional functionality is the way forward. 

Is there more you want to do in this PR?

**Comment by felixdivo** [comment]
I basically updated the tools in `requirements-lint.txt` and now also in `.travis.yml`. I'd wait for the discussion in #951 before doing more, since the macOS build seems to fail for now but there is no need for debugging if we delete anyways.

**Comment by felixdivo** [comment]
Seems like this uncovered [an actual problem](https://github.com/hardbyte/python-can/pull/940/checks?check_run_id=1458088551#step:4:101): https://github.com/hardbyte/python-can/pull/940/checks?check_run_id=1458088551#step:4:101. Pypy support for that lib is aparrently not available, see https://github.com/python/typed_ast/issues/111.

**Comment by felixdivo** [comment]
I also opened #46

**Comment by felixdivo** [comment]
The problem on Python version 3.10 with `python-lazy-object-proxy` should by fixed by `wheel` version 0.36.2.

**Comment by felixdivo** [comment]
Seems like this time, we actually uncovered a bug in our repository: https://github.com/hardbyte/python-can/pull/940/checks?check_run_id=1753568353#step:5:333

**Comment by zariiii9003** [comment]
We could set python 3.10 to experimental and merge this. What do you think?

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error

**Comment by felixdivo** [comment]
@zariiii9003 Is this what you thought about?

**Comment by zariiii9003** [comment]
Yes that's what i meant, thank you. I'm just wondering why the GitHub tests were not run on your last commit. 

Also: the example says `runs-on: ${{ matrix.os }}` while it is `runs-on: ubuntu-latest` in our build.yml. Is this correct?

**Comment by zariiii9003** [comment]
black is incompatible with pypy3. So it fails on step `pip install -r requirements-lint.txt`
Should we separate tests and linters into different jobs? Or is there a way to skip linters only for pypy?

**Comment by felixdivo** [comment]
Ok, so there seem to be more failures...

Somehow efc5f26889a7e593e6d70ed7e486e65e6af1e467 did not disable the `TestMessageSync` test, which is causing problems on GitHub Actions on macOS and Windows. Additionally, there seems to be an issue with the scripts not being available on WIndows.

**Comment by zariiii9003** [comment]
On windows the mypy_extensions are not found during the script-test. I guess the tox environment might not be passed to the subprocess call...

**Comment by felixdivo** [comment]
Thanks @zariiii9003. Somehow that variable was not set despite [this doc]() saying it would be.

The last error seems to be [`File "D:\\a\\python-can\\python-can\\can\\message.py", line 11, in <module>\r\n    from . import typechecking\r\n  File "D:\\a\\python-can\\python-can\\can\\typechecking.py", line 9, in <module>\r\n    import mypy_extensions\r\nModuleNotFoundError: No module named \'mypy_extensions\'`](https://github.com/hardbyte/python-can/pull/940/checks?check_run_id=2331114631#step:5:361). But it also says [`mypy-extensions==0.4.3`](https://github.com/hardbyte/python-can/pull/940/checks?check_run_id=2331114631#step:5:10) is installed right above. I don't really know how tox works though...

**Comment by felixdivo** [comment]
Dependencies now install nicely on PyPy on Windows.

**Comment by felixdivo** [comment]
It's unclear to my, why `test/back2back_test.py::TestThreadSafeBus::test_concurrent_writes` is [failing on macOS, PyPy3](https://github.com/hardbyte/python-can/pull/940/checks?check_run_id=2335703643#step:5:1918). Threading should work in PyPy just like in CPython, where the tests succeeds.

**Comment by zariiii9003** [comment]
Here's a new one (for me): https://github.com/hardbyte/python-can/pull/940/checks?check_run_id=2336410553#step:5:332

And the pypy3 threading error also occurs on windows.

**Comment by felixdivo** [comment]
> Here's a new one (for me): https://github.com/hardbyte/python-can/pull/940/checks?check_run_id=2336410553#step:5:332

Should be fixed. The runners have such an unreliable timing that any kind of deadline/timing assert seems to fail eventually.

**Comment by felixdivo** [comment]
Asyncio error in python 3.10 is tracked in #1005

**Comment by felixdivo** [comment]
@zariiii9003 Everything seems to be done now, right?

**Comment by zariiii9003** [comment]
Depends, do we keep Appveyor?

**Comment by felixdivo** [comment]
It would also touch a few (even more) lines, I'd just do it in a separate PR right after this one is merged. It's already somewhat too large I think.

File: can/ctypesutil.py
Comment (zariiii9003) [inline can/ctypesutil.py:45]: This should be:

```python3
        except AttributeError as attribute_error:
            raise ImportError(
                "Could not map function '{}' from library {}".format(
                    func_name, self._name
                )
            ) from attribute_error
```
Otherwise Traceback is lost, right?

File: can/interface.py
Comment (zariiii9003) [inline can/interface.py:44]: `raise from e`

File: can/interface.py
Comment (zariiii9003) [inline can/interface.py:34]: raise from KeyError reference

File: can/interface.py
Comment (zariiii9003) [inline can/interface.py:54]: `raise from e`

File: can/interfaces/pcan/pcan.py
Comment (zariiii9003) [inline can/interfaces/pcan/pcan.py:635]: Do we know the reason? Is some warning necessary here?

File: test/config.py
Comment (zariiii9003) [inline test/config.py:39]: typo in "may"

File: can/ctypesutil.py
Comment (felixdivo) [reply can/ctypesutil.py:45]: Well, this was intentional, since the message is already quite verbose. If you want to commit your proposal, go ahead. ;)

File: can/interface.py
Comment (felixdivo) [reply can/interface.py:34]: Same as above. This was changed to please pylint (I think it was that one).

File: test/config.py
Comment (felixdivo) [reply test/config.py:39]:
```suggestion
        "only one of IS_APPVEYOR and IS_TRAVIS and GITHUB_ACTIONS may be True at the same time"
```

File: can/ctypesutil.py
Comment (zariiii9003) [reply can/ctypesutil.py:45]: If it's intentional then i trust you there :)

File: can/interfaces/pcan/pcan.py
Comment (felixdivo) [reply can/interfaces/pcan/pcan.py:635]: No, I do not really know what is failing here.

**Review by zariiii9003 — APPROVED** [review]

File: can/interfaces/pcan/pcan.py
Comment (felixdivo) [reply can/interfaces/pcan/pcan.py:630]: Let's just open an issue on this.

Comment (felixdivo) [reply can/interfaces/pcan/pcan.py:630]: Tracked in #1007

**Comment by felixdivo** [comment]
@hardbyte Do you wish to do/have someone do another review or is it OK to be merged as is?

**Comment by hardbyte** [comment]
Good work hunting down all the loose ends! Happy to merge

**Comment by felixdivo** [comment]
@zariiii9003 Nice work, thanks for joining. I'll assemble a much simpler PR that will remove AppVeyor in a minute.

**Pull request merged**
